tools/compile_tf_graph.py

```diff
@@ -612,16 +762,32 @@ def cond(i, net_vars, acc_tas, seq_len_info=None)
       seq_len_info = None
     else:
       i, net_vars, acc_tas, seq_len_info = loop_vars
-      seq_len_info = rec_layer.create_state_vars_recursive(["end_flag", "dyn_seq_len"], seq_len_info)
-    i = rec_layer.create_state_var("i", i)
+
     with tf.name_scope("state"):
       # See _SubnetworkRecCell.GetOutput.body.
       # net_vars is (prev_outputs_flat, prev_extra_flat), and prev_outputs_flat corresponds to self._initial_outputs,
       # where the keys are the layer names.
       # prev_extra is always expected to be batch-major.
-      prev_outputs_data = [self.layer_data_templates[k].output for k in sorted(self._initial_outputs.keys())]
-      net_vars = rec_layer.create_state_vars_recursive(
-        name_prefix="state", initial_values=net_vars, data_shape=(prev_outputs_data, None))
+      prev_outputs_data = []
+      choice_dependent = []
+      for k in sorted(self._initial_outputs.keys()):
+        choice_dependent.append(False)
+        for source_layer in self.net_dict[k]['from']:
+          if source_layer in self.net_dict.keys() and self.net_dict[source_layer]['class'] == ChoiceStateVarLayer.layer_class:
+            choice_dependent[-1] = True
```

Comment:
This code looks very much wrong. There can be *many* different cases how some layer can depend on another layer directly, not just via `from`. And then it can also depend on it indirectly.

Reply:
only direct ones need to be computed after choices are assigned. agree there maybe nicer way to do this dependency check. but all these are state vars which either have to be assigned by Sprint or infered from others

Reply:
I don't understand. Why does it matter for the state var if there is some other layer (or activation function, or whatever) in between or not?
Also, even as a direct layer dependency check, this code is wrong. If the layer is already constructed, you can use `layer.get_dep_layers()`. You should definitely not read from `net_dict`, and hardcode some strange assumptions here. This will definitely easily break for many other configs.

Reply:
@ZhouW321 I still don't understand this. You never answered here.

Reply:
Also, do we really need this?